api/filter_dimensions.go

```diff
@@ -328,6 +349,37 @@ func (api *FilterAPI) checkNewFilterDimension(ctx context.Context, name string,
 	return nil
 }
 
+// createPublicDimensions wraps createPublicDimension for converting arrays of dimensions
+func createPublicDimensions(inputDimensions []models.Dimension) []*models.PublicDimension {
+
+	var outputDimensions []*models.PublicDimension
+	for _, inputDimension := range inputDimensions {
+
+		publicDimension := createPublicDimension(inputDimension)
+		outputDimensions = append(outputDimensions, publicDimension)
+	}
+
+	return outputDimensions
+}
+
+// createPublicDimension creates a PublicDimension struct from a Dimension struct
+func createPublicDimension(dimension models.Dimension) *models.PublicDimension {
```

Comment:
Its worth writing unit test for these non exported functions, `createPublicDimension` and `createPublicDimensions`